### PR TITLE
Allow setting RocksDB log directory

### PR DESCRIPTION
**Pull request by romanz**
Following https://github.com/romanz/electrs/issues/680#issuecomment-1824841535.

### CODE REVIEW

File: internal/config_specification.toml [hunk 1]
```diff
@@ -29,6 +29,12 @@ type = "std::path::PathBuf"
 doc = "Directory to store index database (default: ./db/)"
 default = "\"./db\".into()"
 
+[[param]]
+name = "db_log_dir"
+type = "std::path::PathBuf"
+doc = "Directory to store index database internal log (default: same as specified by `db_dir`)"
+default = "\"\".into()"
```
Comment (Kixunil) [inline internal/config_specification.toml:36]: I think it'd look better if we just made it optional ans only set it if it's `Some`. I'm surprised setting an empty even does the right thing.

**Review by Kixunil — COMMENTED** [review]
Oh, lol, you already did this. Thanks! :D

File: internal/config_specification.toml (hunk 1 above)
Comment (romanz) [reply internal/config_specification.toml:36]: > I'm surprised setting an empty even does the right thing.

I agree, but it seems to work:
https://github.com/facebook/rocksdb/blob/f6fd4b9dbd15dba36f7e5ad23de407b5c26b1460/db/db_impl/db_impl_files.cc#L246

BTW, does `configure_me` support `Optional<PathBuf>`?

Comment (Kixunil) [reply internal/config_specification.toml:36]: Yes, simply leave out `default = ""` and the type in the struct will be wrapped in `Option`. (For mandatory param you write `optional = false`.)

**Pull request merged**